src/app/views/query-response/adaptive-cards/AdaptiveCard.tsx

```diff
@@ -0,0 +1,95 @@
+import * as AdaptiveCardsAPI from 'adaptivecards';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators, Dispatch } from 'redux';
+import { IAdaptiveCardProps } from '../../../../types/adaptivecard';
+import { getAdaptiveCard } from '../../../services/actions/adaptive-cards-action-creator';
+import { Monaco } from '../../common';
+
+class AdaptiveCard extends Component<IAdaptiveCardProps> {
+  private adaptiveCard: AdaptiveCardsAPI.AdaptiveCard;
+  constructor(props: IAdaptiveCardProps) {
+    super(props);
+    this.adaptiveCard = new AdaptiveCardsAPI.AdaptiveCard();
+  }
+
+  public componentDidMount() {
+    const { body , sampleQuery, hostConfig } = this.props;
+    this.props.actions!.getAdaptiveCard(body, sampleQuery);
+    if (hostConfig) {
+      this.adaptiveCard.hostConfig = new AdaptiveCardsAPI.HostConfig(hostConfig);
+    }
+  }
+
+  public componentDidUpdate(nextProps: IAdaptiveCardProps) {
+    const { body , sampleQuery } = this.props;
+    if (JSON.stringify(nextProps.body) !== JSON.stringify(body)) {
+      // we need to update the card as our body has changed
+      this.props.actions!.getAdaptiveCard(body, sampleQuery);
+    }
+  }
+
+  public componentWillUnmount() {
+    // Remove all references
+    delete this.adaptiveCard;
+  }
+
+  public shouldComponentUpdate(nextProps: IAdaptiveCardProps) {
```

Comment:
What do you think of the approach below?

```js
const sameBody = ...
const sameCardData = ...

return !sameBody || !sameCardData
```

Reply:
From my point of view, the implemented approach is a bit easier to read and follow with the code.